packages/account/src/Components/api-token/api-token-delete-button.jsx

```diff
@@ -0,0 +1,101 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Button, Icon, Modal, Text, Popover } from '@deriv/components';
+import { useIsMounted } from '@deriv/shared';
+import { localize } from '@deriv/translations';
+import ApiTokenContext from './api-token-context';
+
+const ApiTokenDeleteButton = ({ token, popoverAlignment = 'left' }) => {
+    const { deleteToken } = React.useContext(ApiTokenContext);
+    const [is_deleting, setIsDeleting] = React.useState(false);
+    const [is_loading, setIsLoading] = React.useState(false);
+    const isMounted = useIsMounted();
+    const timeout_ref = React.useRef(null);
+
+    const getConfirmationBeforeDelete = () => setIsDeleting(true);
+
+    const handleNo = () => setIsDeleting(false);
+
+    const handleYes = () => {
+        setIsLoading(true);
+        deleteToken(token.token).finally(() => {
+            if (isMounted()) {
+                setIsLoading(false);
+                setIsDeleting(false);
+            }
+        });
+    };
+
+    React.useEffect(() => {
+        if (is_deleting) {
+            timeout_ref.current = setTimeout(() => {
+                if (isMounted()) setIsDeleting(false);
+            }, 10000);
+        } else {
```

Comment:
Why do we need this setTimeout? This leads to close the popup by force!

Reply:
Actually such behaviour existed in current behaviour. I've clarified it with usability specialist - we don’t need this timeout anymore. thanks :+1: